01-JS-DOM-APIs/code.js

```diff
@@ -0,0 +1,104 @@
+function show(){
+                let obj = document.getElementById("hidden");
+                if(obj.style.display == "block")
+                        obj.style.display = "none";
+                else
+                        obj.style.display = "block";
+                      }
+
+function eventRequest() {
+                        let xhttp = new XMLHttpRequest();
+                        xhttp.open("GET", "http://api.icndb.com/jokes/random", true);
+                        xhttp.onreadystatechange = function(event) {
+                          let response = JSON.parse(event.target.response);
+                          document.getElementById("hidden").innerHTML = response.value.joke;
+                           if (xhttp.readyState === XMLHttpRequest.DONE) {
+                             if (xhttp.responseText.value) {
+                                 document.getElementById("buttonSection").innerHTML  = xhttp.responseText.value.joke;
+                             }
+                           }
+                       }
+                      xhttp.send();
+                    }
+/*Ejercicio 7*/
+function eventOnClick(){
+  let config = {
+    url: 'http://api.icndb.com/jokes/random'
+  };
+
+  let xhttp = new XMLHttpRequest();
+
+  function resolve(xhttp) {
+
+     let response = JSON.parse(event.target.response);
+
+      document.getElementById("hidden").innerHTML = response.value.joke;
+
+      if (xhttp.readyState === XMLHttpRequest.DONE) {
+        if (xhttp.responseText.value) {
+          document.getElementById("buttonSection").innerHTML  = xhttp.responseText.value.joke;
+        }
+      }
+  }
+
+  function reject() {
+     document.getElementById("hidden").style.backgroundColor="red";
+  }
+
+  eventRequestReusable(config).then(resolve,reject);
+};
+
+function eventRequestReusable(object) {
+  let promise = new Promise( function (resolve, reject) {
+    let xhttp = new XMLHttpRequest();
+
+    xhttp.open("GET", object.url);
+    xhttp.send();
+    xhttp.onload = function () {
+      if (this.status == 200) {
+        resolve(xhttp.response);
+      } else {
+        reject(this.statusText);
+      }
+    };
+    xhttp.onerror = function () {
+     reject(this.statusText);
+    };
+  });
+  return promise;
+};
+
+/*Ejercicio 9*/
+function load () {
+  console.log('load');
+  let changeElement = document.getElementById("changeclass");
+  changeElement.style.transition = "all 3s ease-in-out";
+  changeElement.style.opacity = 1;
+}
+
+function configAjax (methodHttp, url, asyncronic){
+  this.methodHttp = methodHttp;
```

Comment:
this is not right, *this.methodHttp = methodHttp* is assigning a methodHttp variable to the global scope, this could cause serious complications, beware of how you apply scope variables